src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -447,6 +456,23 @@ public boolean isCredentialValid() throws IOException {
     }
 
     /**
+     * Ensures that the API URL is valid.
+     */
+    public boolean isApiUrlValid() throws IOException {
+        try {
+            GHApiInfo apiInfo = retrieve().to("/", GHApiInfo.class);
+            try {
+                new URL(apiInfo.getRateLimitUrl());
```

Comment:
The same way than if disappear other attributes. This method will return `null` is this case. By the way, when it happens GitHub API v4 will be published and a review of this client will be required.